docs/src/heat_equation.md

```diff
@@ -0,0 +1,217 @@
+
+Consider the heat equation
+
+```math
+\partial_t u(t,x) = \mu \partial_x^2 u(t,x),\quad u(0,x)=u_0(x),
+```
+
+with ``μ ≥ 0``, ``t≥ 0``, ``x\in[0,1]``, and homogeneous Neumann boundary conditions.
+We choose the classical central finite difference discretization of the
+Laplacian, resulting in the ODE
+
+```math
+\partial_t u(t) = L u(t),
+\quad
+L = \frac{\mu}{\Delta x^2} \begin{pmatrix}
+    -1 & 1 \\
+    1 & -2 & 1 \\
+    & \ddots & \ddots & \ddots \\
+    && 1 & -2 & 1 \\
+    &&& 1 & -1
+\end{pmatrix}.
+```
+
+The system can be written as a conservative PDS with production terms
+
+```math
+\begin{aligned}
+&p_{i,i-1}(t,\mathbf u(t)) = \frac{\mu}{\Delta x^2} u_{i-1}(t),\quad i=2,\dots,N, \\
+&p_{i,i+1}(t,\mathbf u(t)) = \frac{\mu}{\Delta x^2} u_{i-1}(t),\quad i=1,\dots,N-1,
+\end{aligned}
+```
+
+and destruction terms ``d_{i,j} = p_{j,i}``.
+
+
+## Solution of the production-destruction system
+
+Now we are ready to define a `ConservativePDSProblem` and to solve this
+problem with a method of
+[PositiveIntegrators.jl](https://github.com/SKopecz/PositiveIntegrators.jl) or
+[OrdinaryDiffEq.jl](https://docs.sciml.ai/OrdinaryDiffEq/stable/).
+In the following we use ``N=100`` nodes and the time domain ``t\in[0,1]``.
+Moreover, we choose the initial condition
+
+```math
+u_0(x) = \cos(\pi x).
+```
+
+```@example HeatEquation
+x_boundaries = range(0, 1, length = 101)
+x = x_boundaries[1:end-1] .+ step(x_boundaries) / 2
+u0 = @. cospi(x) # initial solution
+tspan = (0.0, 1.0) # time domain
+
+nothing #hide
+```
+
+We will choose three different matrix types for the production terms and
+the resulting linear systems:
+
+1. standard dense matrices (default)
+2. sparse matrices (from SparseArrays.jl)
+3. tridiagonal matrices (from LinearAlgebra.jl)
+
+
+### Standard dense matrices
+
+```@example HeatEquation
+using PositiveIntegrators # load ConservativePDSProblem
+
+function heat_eq_P!(P, u, μ, t)
+    fill!(P, 0)
+    N = length(u)
+    Δx = 1 / N
+    μ_Δx2 = μ / Δx^2
+
+    let i = 1
+        # Neumann boundary condition
+        P[i, i + 1] = u[i + 1] * μ_Δx2
+    end
+
+    for i in 2:(length(u) - 1)
+        # interior stencil
+        P[i, i - 1] = u[i - 1] * μ_Δx2
+        P[i, i + 1] = u[i + 1] * μ_Δx2
+    end
+
+    let i = length(u)
+        # Neumann boundary condition
+        P[i, i - 1] = u[i - 1] * μ_Δx2
+    end
+
+    return nothing
+end
+
+μ = 1.0e-2
+prob = ConservativePDSProblem(heat_eq_P!, u0, tspan, μ) # create the PDS
+
+sol = solve(prob, MPRK22(1.0); save_everystep = false)
+
+nothing #hide
+```
+
+```@example HeatEquation
+using Plots
+
+plot(x, u0; label = "u0", xguide = "x", yguide = "u")
+plot!(x, last(sol.u); label = "u")
+```
+
+
+### Sparse matrices
+
+To use different matrix types for the production terms and linear systems,
+you can use the keyword argument `p_prototype` of
+[`ConservativePDSProblem`](@ref) and [`PDSProblem`](@ref).
+
+```@example HeatEquation
```

Comment:
Using a sparse matrix is even slower than using a dense matrix. Maybe $N=100$ is to small?

Reply:
The improvements for sparse matrices have not been included in the built of the docs you looked at. I merged `main` into this branch, so it should be better now.